src/js/tech/tech.js

```diff
@@ -805,6 +806,12 @@ Tech.withSourceHandlers = function(_Tech){
     this.disposeSourceHandler();
     this.off('dispose', this.disposeSourceHandler);
 
+    // if we have a source and get another one
```

Comment:
it seems like it should I will have to work on adding a test for that case, if you think that its a potential scenario